pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -275,6 +276,20 @@ func checkNetworkManagerIsRunning() error {
 }
 
 func checkSystemdResolvedIsRunning() error {
+	resolvFilePath := "/etc/resolv.conf"
+	// Check /ETC/RESOLV.CONF section in `man systemd-resolved`
+	systemdResolvedManageResolveFilePath := []string{"/run/systemd/resolve/stub-resolv.conf",
+		"/usr/lib/systemd/resolv.conf",
+		"/run/systemd/resolve/resolv.conf"}
+	// Check if /etc/resolv.conf is managed by systemd-resolved
+	rFilePath, err := filepath.EvalSymlinks(resolvFilePath)
+	logging.Debugf("resolve conf file path: %s", rFilePath)
+	if err != nil {
+		return err
+	}
+	if !slices.Contains(systemdResolvedManageResolveFilePath, rFilePath) {
```

Comment:
I'd use https://github.com/crc-org/crc/blob/3956e85655b4e2b6ebefd52aea7417566018fa34/pkg/strings/strings.go#L8-L15 especially as the `slices` package is 1.21 only. 
No strong objections to using this package though, but this should be done in a commit removing our implementation of this functionality.

Reply:
Yes. sure I also don't want to wait it for 1.21 to land first and then this PR, will update it to use this one.

Reply:
Done.